src/Microsoft.AspNet.Mvc.ModelBinding/Binders/MutableObjectModelBinder.cs

```diff
@@ -90,16 +95,7 @@ protected virtual bool CanUpdateProperty(ModelMetadata propertyMetadata)
                 return true;
             }
 
-            // 3. The model name is not prefixed and a value provider can directly provide a value for the model name.
```

Comment:
Aa I see, 
so basically when we fallback on empty string, this check caused non top level model to get created even though there was nothing which was binding a value? I am sorry This is confusing, it would be great if you can walk me through an example.